// eas-cli/build/commandUtils/new/commands.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.initializeGitRepositoryAsync = exports.installProjectDependenciesAsync = exports.cloneTemplateAsync = void 0;
const tslib_1 = require("tslib");
const fs_extra_1 = tslib_1.__importDefault(require("fs-extra"));
const path_1 = tslib_1.__importDefault(require("path"));
const log_1 = tslib_1.__importDefault(require("../../log"));
const git_1 = require("../../onboarding/git");
const installDependencies_1 = require("../../onboarding/installDependencies");
const runCommand_1 = require("../../onboarding/runCommand");
async function cloneTemplateAsync(targetProjectDir) {
    const githubUsername = 'expo';
    const githubRepositoryName = 'expo-template-default';
    log_1.default.log(`📂 Cloning the project to ${targetProjectDir}`);
    log_1.default.newLine();
    const cloneMethod = (await (0, git_1.canAccessRepositoryUsingSshAsync)({
        githubUsername,
        githubRepositoryName,
    }))
        ? 'ssh'
        : 'https';
    const { targetProjectDir: finalTargetProjectDirectory } = await (0, git_1.runGitCloneAsync)({
        githubUsername,
        githubRepositoryName,
        targetProjectDir,
        cloneMethod,
    });
    return finalTargetProjectDirectory;
}
exports.cloneTemplateAsync = cloneTemplateAsync;
async function installProjectDependenciesAsync(projectDir, packageManager) {
    await (0, installDependencies_1.installDependenciesAsync)({
        outputLevel: 'none',
        projectDir,
        packageManager,
    });
    const dependencies = ['expo-updates', '@expo/metro-runtime'];
    for (const dependency of dependencies) {
        await (0, runCommand_1.runCommandAsync)({
            cwd: projectDir,
            command: 'npx',
            args: ['expo', 'install', dependency],
            showOutput: false,
        });
    }
}
exports.installProjectDependenciesAsync = installProjectDependenciesAsync;
async function initializeGitRepositoryAsync(projectDir) {
    await fs_extra_1.default.remove(path_1.default.join(projectDir, '.git'));
    const commands = [['init'], ['add', '.'], ['commit', '-m', 'Initial commit']];
    for (const args of commands) {
        await (0, runCommand_1.runCommandAsync)({
            cwd: projectDir,
            command: 'git',
            args,
            showOutput: false,
        });
        log_1.default.log();
    }
}
exports.initializeGitRepositoryAsync = initializeGitRepositoryAsync;
